remotes/docker/resolver.go

```diff
@@ -182,6 +194,10 @@ func (r *dockerResolver) Resolve(ctx context.Context, ref string) (string, ocisp
 			return "", ocispec.Descriptor{}, err
 		}
 
+		for key, val := range r.headers {
+			req.Header.Set(key, val)
+		}
+
 		// set headers for all the types we support for resolution.
 		req.Header.Set("Accept", strings.Join([]string{
```

Comment:
now that you have a generic headers map, seems like this `Accept` header could be part of that `NewResolver` initialization, and with a similar flow to `User-Agent`, would even allow uses of the resolver to set a unique `Accept` list, which isn't possible in the current code.

Reply:
I had the same thoughts, but being new to this project, I wasn't entirely sure if potentially overriding the default `Accept` list was an intended use-case.